File: Hortsman/src/Part5/Car.scala
package Part5

class Car(val mark: String, val model: String) {
  private var _year = -1
  var registrationNumber = ""

  def this(mark: String, model: String, year: Int) {
    this(mark, model)
    this._year = year
  }

  def this(mark: String, model: String, registrationNumber: String) {
    this(mark, model)
    this.registrationNumber = registrationNumber
  }

  def this(mark: String, model: String, year: Int, registrationNumber: String) {
    this(mark, model, year)
    this.registrationNumber = registrationNumber
  }

  def year = _year
}

File: Hortsman/src/bin/src/Part5/Car.cs
public class Car
{
    public string Mark { get; private set; }
    public string Model { get; private set; }
    public int Year { get; private set; } = -1;
    public string RegistrationNumber { get; set; } = "";

    public Car(string mark, string model)
    {
        Mark = mark;
        model = model;
    }

    public Car(string mark, string model, int year) : base(mark, model)
    {
        Year = year
    }

    public Car(string mark, string model, string registrationNumber) : base(mark, model)
    {
        RegistrationNumber = registrationNumber
    }

    public Car(string mark, string model, int year, string registrationNumber): base(mark, model, year)
    {
        RegistrationNumber = registrationNumber
    }
}
